### ISSUE TITLE
Block recipe_type submission if names match

### ISSUE THREAD

**Issue by mikenholt** [ngageoint/scale#1710]
**Pain Point? Please describe.**
When a recipe with an existing name is submitted to `recipe-types/` Scale appends some letters or numbers and allows it to go though.  This can be annoying if you accidentally try to update a recipe at this endpoint since you can't ever delete that recipe.  

**Desired Solution**
Block any requests to `recipe_types/` if the name matches an existing name. Return an error that states the recipe needs a new name or they need to patch the recipe at `recipe-types/{name}`

**Additional Context**
`job-types/` behavior should be similar.

**Comment by JohnPTobe** [comment]
I agree on blocking duplicate names for recipe types.  The job-types api doesn't take names but instead takes seed manifests which contain a name and version.  If the name/version combo is unique it creates a new job type with that name and version.  If not, it creates a new revision of that job type.  This causes a potential issue with images from multiple repositories that share the same name/version.  Within a docker repository, seed images are guaranteed to be unique but seed images from multiple repositories could have collisions.  This would need to be addressed in the future.

**Comment by mikenholt** [comment]
I am moving this off Sprint 3.  I looked at the code for recipes and there are already checks in place to block recipe name collisions.  From what @JohnPTobe the job-type fix will have to be more sophisticated.
